feat(ramp): offramp analytics

georgeweiler opened the pull request:
## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**
N/A - we do not track analytics issues here in github

## **Manual testing steps**
Check analytics event logs and confirm the sell events are triggered

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

georgeweiler committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-mobile into feat/off-ramp

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/off-ramp

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/off-ramp

georgeweiler committed:
refactor (ramp): updates ramp analytics interfaces to include sell events

georgeweiler committed:
chore: add new offramp interfaces for future use

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
chore: adds SELL_BUTTON_CLICKED analytics

wachunei commented:
Please mark as Draft until PR is ready for review and all checks are passing 🙏

georgeweiler committed:
chore: adds OFFRAMP_REGION_UNSUPPORTED event

georgeweiler committed:
chore: adds OFFRAMP_REGION_RESET event

georgeweiler committed:
chore: adds OFFRAMP_PAYMENT_METHOD_SELECTED event

georgeweiler committed:
chore: adds OFFRAMP_CANCELLED event

georgeweiler committed:
chore: adds OFFRAMP_QUOTES_RECEIVED event

georgeweiler committed:
chore: adds OFFRAMP_PROVIDER_SELECTED event

georgeweiler committed:
chore: adds OFFRAMP_PROVIDER_DETAILS_VIEWED event

georgeweiler committed:
chore: adds OFFRAMP_DIRECT_PROVIDER_CLICKED event

georgeweiler committed:
chore: adds OFFRAMP_PURCHASE_SUBMITTED event

georgeweiler committed:
chore: adds the remaining offramp events

georgeweiler committed:
chore: updates the purchase completed, cancelled, failed events

georgeweiler committed:
refactor: prefer amount over crypto_amount for analytics payloads

georgeweiler committed:
chore (ramp): cleans up analytics, leaves a few comments

wachunei commented on app/components/UI/Ramp/buy/Views/PaymentMethods/PaymentMethods.tsx:
```diff
@@ -100,14 +108,19 @@ const PaymentMethods = () => {
-      location: 'Payment Method Screen',
-    });
+    trackEvent(
+      rampType === RampType.BUY
+        ? 'ONRAMP_CONTINUE_TO_AMOUNT_CLICKED'
+        : 'OFFRAMP_CONTINUE_TO_AMOUNT_CLICKED',
```
use `isBuy` or `isSell` instead.

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:
```diff
@@ -124,32 +128,55 @@ function Quotes() {
+      location: 'Quotes Screen' as ScreenLocation,
+    };
+
+    if (rampType === RampType.BUY) {
```
Let's adopt `isBuy`/`isSell` here too.

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:
```diff
@@ -167,12 +194,19 @@ function Quotes() {
-          provider_onramp: quote.provider.name,
-        });
+
+        if (rampType === RampType.BUY) {
```
Let's adopt `isBuy`/`isSell` here too.

wachunei commented on app/components/UI/Ramp/buy/Views/Regions/Regions.tsx:
```diff
@@ -63,11 +63,18 @@ const RegionsView = () => {
-      chain_id_destination: selectedChainId,
-    });
-  }, [selectedChainId, trackEvent]);
+    if (rampType === RampType.BUY) {
```
Let's adopt `isBuy`/`isSell` here.

wachunei commented on app/components/UI/Ramp/common/Views/NetworkSwitcher/NetworkSwitcher.tsx:
```diff
@@ -93,11 +94,18 @@ function NetworkSwitcher() {
-      chain_id_destination: selectedChainId,
-    });
-  }, [selectedChainId, trackEvent]);
+    if (rampType === RampType.BUY) {
```
Let's adopt `isBuy`/`isSell` here too.

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:
```diff
@@ -354,46 +401,81 @@ function Quotes() {
-          }),
-        );
+          };
+          if (rampType === RampType.BUY) {
```
Same comment about `isBuy`/`isSell` 😅 .

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:
```diff
@@ -354,46 +401,81 @@ function Quotes() {
-        });
+        };
+
+        if (rampType === RampType.BUY) {
```
Same comment about `isBuy`/`isSell` 😅 .

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:
```diff
@@ -185,22 +219,35 @@ function Quotes() {
-        });
+        };
+
+        if (rampType === RampType.BUY) {
```
Same comment about `isBuy`/`isSell` 😅 .

wachunei commented on app/components/UI/Ramp/common/Views/OrderDetails.tsx:
```diff
@@ -62,15 +63,26 @@ const OrderDetails = () => {
+      };
+      if (order.orderType === OrderOrderTypeEnum.Buy) {
+        trackEvent('ONRAMP_PURCHASE_DETAILS_VIEWED', {
+          ...payload,
+          provider_onramp: (order.data as Order)?.provider.name,
```
`(order.data as Order)?.provider.name` is being accessed twice, I propose moving it to a variable next to `payload`.

wachunei commented on app/components/UI/Ramp/common/Views/Settings/AddActivationKey.tsx:
```diff
@@ -6,7 +6,6 @@ import Text from '../../../../../Base/Text';
 import StyledButton from '../../../../StyledButton';
 import Row from '../../components/Row';
 import ScreenLayout from '../../components/ScreenLayout';
-
```
Undo this change since is not relevant to the PR

wachunei commented on app/components/UI/Ramp/common/Views/Settings/Settings.tsx:
```diff
@@ -39,11 +41,16 @@ function Settings() {
-      location: 'Settings Screen',
-    });
+    trackEvent(
+      rampType === RampType.BUY
```
Let's adopt `isBuy`/`isSell` instead of `RampType` comparison.

wachunei commented on app/components/UI/Ramp/common/components/ErrorView.tsx:
```diff
@@ -96,14 +96,15 @@ function ErrorView({
 
   useEffect(() => {
-    trackEvent('ONRAMP_ERROR', {
+    trackEvent(`${isBuy ? 'ON' : 'OFF'}RAMP_ERROR`, {
```
This is correct but a little bit too complex for the future, let's change it to the approach seen in other files 
```suggestion
    trackEvent(isBuy ? 'ONRAMP_ERROR' : 'OFFRAMP_ERROR', {
```

wachunei commented on app/components/UI/Ramp/common/components/InfoAlert.tsx:
```diff
@@ -70,41 +71,42 @@ const InfoAlert: React.FC<Props> = ({
     (url: string) => {
       Linking.openURL(url);
-      trackEvent('ONRAMP_EXTERNAL_LINK_CLICKED', {
+      trackEvent(`${isBuy ? 'ON' : 'OFF'}RAMP_EXTERNAL_LINK_CLICKED`, {
```
Same comment about doing ternary between the whole strings instead, this is correct but maybe too complex

wachunei commented on app/components/UI/Ramp/common/components/InfoAlert.tsx:
```diff
@@ -70,41 +71,42 @@ const InfoAlert: React.FC<Props> = ({
     (url: string) => {
       Linking.openURL(url);
-      trackEvent('ONRAMP_EXTERNAL_LINK_CLICKED', {
+      trackEvent(`${isBuy ? 'ON' : 'OFF'}RAMP_EXTERNAL_LINK_CLICKED`, {
```
Same comment about doing ternary between the whole strings instead, this is correct but maybe too complex

wachunei commented on app/components/UI/Ramp/common/components/InfoAlert.tsx:
```diff
@@ -70,41 +71,42 @@ const InfoAlert: React.FC<Props> = ({
     (url: string) => {
       Linking.openURL(url);
-      trackEvent('ONRAMP_EXTERNAL_LINK_CLICKED', {
+      trackEvent(`${isBuy ? 'ON' : 'OFF'}RAMP_EXTERNAL_LINK_CLICKED`, {
```
Same comment about doing ternary between the whole strings instead, this is correct but maybe too complex

wachunei commented on app/components/UI/ReceiveRequest/index.js:
```diff
@@ -193,6 +193,7 @@ class ReceiveRequest extends PureComponent {
       toggleReceiveModal();
       navigation.navigate(Routes.RAMP.BUY);
       InteractionManager.runAfterInteractions(() => {
+        // ASK PEDRO - do we need an "onSell" method?
```
Yes, we need `SELL_BUTTON_CLICK` but attached to the Sell button (currently there's only one)

wachunei commented on app/components/UI/Ramp/common/types/analytics.ts:
```diff
@@ -10,167 +13,340 @@ export type ScreenLocation =
+  // after redirection events will go here
+}
+
+export type AnalyticsEvents = {
+  [K in keyof typeof MetaMetricsEvents]: K extends keyof EventTypes
+    ? EventTypes[K]
+    : never;
+};
```
We must undo this change. This file must export only two interfaces:
`ScreenLocation` and `AnalyticsEvents`, like before.

MetaMetricsEvents is an object coupled to the current mobile analytics implementation, the `AnalyticsEvents` interface exported here must remain agnostic to the current implementation. 

This means we must not export the smaller interfaces either.

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -30,7 +28,15 @@ import processOrder from './common/orderProcessor';
 import { aggregatorOrderToFiatOrder } from './common/orderProcessor/aggregator';
 import { trackEvent } from './common/hooks/useAnalytics';
-import { AnalyticsEvents } from './common/types';
+import {
+  OffRampPurchaseCanceled,
+  OffRampPurchaseCompleted,
+  OffRampPurchaseFailed,
+  OnRampPurchaseCanceled,
+  OnRampPurchaseCompleted,
+  OnRampPurchaseFailed,
+  RampPurchaseCompleted,
+} from './common/types';
```
These exports/imports must be removed and use `import { AnalyticsEvents } from './common/types';` instead, you can access the type by using the key string.

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -85,52 +60,94 @@ export const getAggregatorAnalyticsPayload = (
-        | 'ONRAMP_PURCHASE_FAILED'
-        | 'ONRAMP_PURCHASE_CANCELLED'
-        | 'ONRAMP_PURCHASE_COMPLETED']
+    | OnRampPurchaseFailed
+    | OnRampPurchaseCanceled
+    | OnRampPurchaseCompleted
+    | OffRampPurchaseFailed
+    | OffRampPurchaseCanceled
+    | OffRampPurchaseCompleted
     | null
```
Follow the same pattern as before.

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -85,52 +60,94 @@ export const getAggregatorAnalyticsPayload = (
     case FIAT_ORDER_STATES.FAILED: {
-      return ['ONRAMP_PURCHASE_FAILED', failedOrCancelledParams];
+      return [
+        `${isBuy ? 'ON' : 'OFF'}RAMP_PURCHASE_FAILED`,
```
Same comment about doing ternary between the whole strings instead, this is correct but maybe too complex.

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -85,52 +60,94 @@ export const getAggregatorAnalyticsPayload = (
     case FIAT_ORDER_STATES.CANCELLED: {
-      return ['ONRAMP_PURCHASE_CANCELLED', failedOrCancelledParams];
+      return [
+        `${isBuy ? 'ON' : 'OFF'}RAMP_PURCHASE_CANCELLED`,
```
Same comment about doing ternary between the whole strings instead, this is correct but maybe too complex.

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -85,52 +60,94 @@ export const getAggregatorAnalyticsPayload = (
     case FIAT_ORDER_STATES.COMPLETED: {
-      return ['ONRAMP_PURCHASE_COMPLETED', completedPayload];
+      return [
+        `${isBuy ? 'ON' : 'OFF'}RAMP_PURCHASE_COMPLETED`,
```
Same comment about doing ternary between the whole strings instead, this is correct but maybe too complex

wachunei commented on app/components/UI/Tokens/index.tsx:
```diff
@@ -313,6 +313,7 @@ const Tokens: React.FC<TokensI> = ({ tokens }) => {
     );
   };
 
+  // ASK PEDRO - do we need a "goToSell" method?
```
Not here at the moment, there's no `Sell` button planned to be here at the moment

wachunei commented on app/components/Views/Asset/index.js:
```diff
@@ -477,6 +477,7 @@ class Asset extends PureComponent {
     const onBuy = () => {
       navigation.navigate(Routes.RAMP.BUY);
       InteractionManager.runAfterInteractions(() => {
+        // ASK PEDRO - go to
```
There's no current plan to have a sell button here at the moment, remove comment 🙏

wachunei commented on app/components/Views/SendFlow/SendTo/index.js:
```diff
@@ -317,6 +317,7 @@ class SendFlow extends PureComponent {
     this.setState({ toInputHighlighted: !toInputHighlighted });
   };
 
+  // ASK PEDRO - do we need a go to sell
```
There's no current plan to have a sell button here at the moment, remove comment 🙏

wachunei commented on app/components/Views/WalletActions/WalletActions.tsx:
```diff
@@ -80,13 +79,14 @@ const WalletActions = () => {
+      Analytics.trackEventWithParameters(
+        MetaMetricsEvents.SELL_BUTTON_CLICKED,
+        {
+          text: 'Buy',
```
Let's change it to `Sell`

wachunei reviewed (changes requested):
Changes requested, main changes requested are: use `isBuy`/`isSell` instead of RampType and revert our analytics internal interface to be decoupled of current MMetrics implementation. I did not see any changes related to tests, we must create tests to confirm the off-ramp methods are being called when `isBuy` is false (you can do this by mocking the sdk hook)

Please use the conventional commit on the PR title (eg: `feat(ramp): <description>`, `refactor(ramp): <description>` and **set is as draft** instead of using `WIP` in the title.

georgeweiler committed:
chore (ramp): clean code updates from code review suggestions

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
test(ramp): updates payment method test to include new analytics events

georgeweiler committed:
test(ramp): updates getStarted test to include new analytics events

georgeweiler committed:
test(ramp): updates Region test to include new analytics events

georgeweiler committed:
test(ramp): adds tests for BuildQuote and Quote view analytics

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
chore(ramp): resolves a linting issue

georgeweiler committed:
Merge branch 'feat/off-ramp' into feat/offramp-analytics

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
Merge branch 'feat/offramp-analytics' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
chore(ramp): clean up analytics interfaces to depend on a single export

georgeweiler committed:
test(ramp): update NetworkSwitcher component to include buy & sell copy changes

wachunei commented:
I'll convert back to draft until CI is passing

georgeweiler committed:
chore(ramp): copy updates to buildQuote to support sell feature

georgeweiler committed:
chore(ramp): updates getStarted test to reset mockstate before asserting

georgeweiler committed:
chore(ramp): small updates to improve paymentMethods test

georgeweiler committed:
chore(ramp): small updates to improve Quotes test

georgeweiler committed:
chore(ramp): updates test to reset mockstate before asserting

georgeweiler committed:
feat(ramp): adds OFFRAMP_REGION_RESET event

georgeweiler committed:
chore(ramp): analytics payload updates for ramp components

georgeweiler committed:
chore(ramp): analytics fixes and clean up for ramp common components

georgeweiler committed:
chore(ramp): removes offramp_external_link_clicked

georgeweiler committed:
chore(ramp): adds back missing ONRAMP_EXTERNAL_LINK_CLICKED event name

georgeweiler committed:
chore(ramp): fixes linting errors on analytics branch

georgeweiler committed:
chore(ramp): remove typo

georgeweiler committed:
chore(ramp): fixes linting errors on analytics branch

georgeweiler committed:
test(ramp): update paymentMethod test to use inline snapshots instead of mocked payload variable

georgeweiler committed:
Merge branch 'feat/off-ramp' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
Merge branch 'feat/off-ramp' into feat/offramp-analytics

georgeweiler committed:
fix(ramp): fixes duplicate variable issue

georgeweiler committed:
test(ramp): fixes variable capitalization on region test

georgeweiler committed:
refactor(ramp): merges the offramp and onramp region selected events

georgeweiler committed:
refactor(ramp): fixes region test

georgeweiler committed:
chore(ramp): rm console.log

georgeweiler committed:
feat(ramp): adds welcome page event for ramps

georgeweiler committed:
fix(ramp): fixes the source/destination definitions for ramp analytics (#7831)

## **Description**
This fixes the incorrect assignment of `currency_source` and
`currency_destination` definitions in the offramp analytics payloads.

## **Related issues**
N/A

## **Manual testing steps**
Check logs when analytics trigger and verify the source/destination
values

## **Screenshots/Recordings**
N/A

### **Before**
N/A

### **After**
N/A

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.test.tsx:
```diff
@@ -162,6 +164,7 @@ describe('Quotes', () => {
   });
 
   it('navigates and tracks event on cancel button press', async () => {
+    // test for buy event
```
Remove this comment, if you want to explicitly note this is for buy, then rename the test, for example: `navigates and tracks event on cancel button press for buy flow`

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.test.tsx:
```diff
@@ -174,11 +177,27 @@ describe('Quotes', () => {
     });
   });
+  it('navigates and tracks event on SELL cancel button press', async () => {
+    // test for offramp cancel event
```
Remove this comment, if you think the test needs more context then improve the test description

wachunei commented on app/components/UI/Ramp/buy/Views/Quotes/Quotes.test.tsx:
```diff
@@ -471,9 +526,63 @@ describe('Quotes', () => {
   });
 
+  it('calls track event on sell quotes received and sell quote error', async () => {
+    // test for offramp cancel event
```
Remove this comment, if you think the test needs more context then improve the test description

wachunei commented on app/components/UI/Ramp/common/types/analytics.ts:
```diff
@@ -10,167 +11,321 @@ export type ScreenLocation =
   | 'Settings Screen';
 
+interface RampButtonClicked {
+  text?: 'Buy' | 'Buy Native Token';
```
Any reason why this is optional now? Every button of the ramp flow is either a generic "Buy"  or "Buy Native Token". I think we are missing the "Sell" case?

georgeweiler replied:
it's for the `*_GET_STARTED_CLICKED` events. Made it optional so I could reuse the interface. Happy to make another one if you need it's needed

wachunei replied:
I see, it is ok to have the same interface but I'd prefer then to add the 'Get Started' case to the text and keep it required.

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -85,52 +52,100 @@ export const getAggregatorAnalyticsPayload = (
     exchange_rate:
       (Number(fiatOrder.amount) - Number(fiatOrder.fee)) /
       Number(fiatOrder.cryptoAmount),
+    // gas fee,
+    // processing fee
```
remove these comments or add a proper `// TODO: <description>` comment

wachunei commented on app/components/UI/Ramp/index.tsx:
```diff
@@ -85,52 +52,100 @@ export const getAggregatorAnalyticsPayload = (
     }
     case FIAT_ORDER_STATES.COMPLETED: {
-      return ['ONRAMP_PURCHASE_COMPLETED', completedPayload];
+      return [
+        isBuy ? 'ONRAMP_PURCHASE_COMPLETED' : 'OFFRAMP_PURCHASE_COMPLETED',
+        isBuy ? buyCompletePayload : sellCompletePayload,
+      ];
```
refactor to a single branching condition
```suggestion
      return isBuy
        ? ['ONRAMP_PURCHASE_COMPLETED', buyCompletePayload]
        : ['OFFRAMP_PURCHASE_COMPLETED', sellCompletePayload];
```

georgeweiler committed:
chore(ramp): some small code cleanup after review

georgeweiler committed:
Merge branch 'feat/offramp-analytics' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
Merge branch 'feat/off-ramp' into feat/offramp-analytics

georgeweiler committed:
refactor(ramp): optimizes some ramp analytics event names

georgeweiler committed:
Merge branch 'feat/offramp-analytics' of https://github.com/MetaMask/metamask-mobile into feat/offramp-analytics

georgeweiler committed:
chore(ramp): adds Get Started text to welcome page event

lorenzosantos commented:
QA'd in simulator--LGTM!

wachunei commented on app/components/UI/Ramp/common/components/RegionModal.tsx:
```diff
@@ -140,22 +140,24 @@ const RegionModal: React.FC<Props> = ({
-        country_onramp_id: regionInTransit?.id ?? region.id,
-        state_onramp_id: regionInTransit ? region.id : undefined,
+
+      trackEvent(`RAMP_REGION_SELECTED`, {
```
Use single quotes
```suggestion
      trackEvent('RAMP_REGION_SELECTED', {
```

wachunei reviewed (commented):
One little change

georgeweiler committed:
chore(ramp): use single quotes for event name

wachunei reviewed (approved):
LGTM!
